Create Flexbox-based Homepage Layout

Aeeshadam opened the pull request:
This PR implements a basic homepage layout using Flexbox for responsiveness and modern design principles. All elements are styled using CSS Flexbox

Aeeshadam committed:
created a page layout with flexbox

Aeeshadam committed:
removed unused class

Aeeshadam committed:
added max width and spacing variables

aayusachdev commented on src/App.js:
```diff
@@ -0,0 +1,6 @@
+import Home from "./Home/index.js";
```
While importing from components, you don't have to put `index.js` in the import line.

aayusachdev commented on src/App.js:
```diff
@@ -0,0 +1,6 @@
+import Home from "./Home/index.js";
+
+const App = () => {
```
single line arrow function can be written as an expression.

aayusachdev commented on src/Home/Footer.js:
```diff
@@ -0,0 +1,27 @@
+    <footer className={styles.footer}>
+      <div className={styles.footerColumns}>
+        {footerLinks.map((category) => (
+          <ul className={styles.footerLinks} key={category.title}>
```
is `category.title` a unique and stable constant here. If not, what other alternatives can we have here?

aayusachdev commented on src/Home/Footer.js:
```diff
@@ -0,0 +1,27 @@
+        {footerLinks.map((category) => (
+          <ul className={styles.footerLinks} key={category.title}>
+            {category.links.map((link) => (
+              <li key={link.title}>
```
same comment as for `category.title`.

aayusachdev commented on src/Home/Footer.js:
```diff
@@ -0,0 +1,27 @@
+  );
+};
+
+export default Footer;
```
What is the use of writing export default in the end here? Why not define it directly on Line 5?

aayusachdev commented on src/Home/Hero.js:
```diff
@@ -0,0 +1,25 @@
+          at aperiam nesciunt, beatae voluptatem laudantium eveniet odit fuga
+          earum tenetur ad?
+        </p>
+        <button>
```
Why are we wrapping an anchor with a button here?

aayusachdev commented on src/Home/Navbar.js:
```diff
@@ -0,0 +1,23 @@
+        </a>
+      </div>
+      <ul className={styles.navLinks}>
+        {links.map((link) => (
```
what if `links` here is undefined, this mapping will fail and result in a WSOD. What can we do here, to prevent it?

aayusachdev commented on src/Home/Services.js:
```diff
@@ -0,0 +1,38 @@
+      <section className={styles.serviceContainer}>
+        <h2>Learn More About Our Services</h2>
+        <div className={styles.cardContainer}>
+          {serviceData.map((service) => (
```
what if `serviceData` is undefined here, it will result in WSOD. What can we do to prevent such errors in production?

aayusachdev commented:
Hey @Aeeshadam i have completed the review now. Here is the PR feedback-

It looks like a good attempt, good work. I've added few comments, can you please address them.

Your commit messages could be more clearer. To help here, you can read about [conventional code commits](https://conventionalcomments.org/). Let me know we can discuss this more

Good job once again!:)

Aeeshadam committed:
refactor: restructure app and implement code review suggestions

- Reorganized file structure and updated naming conventions for better clarity
- Footer: destructured  and  objects; added optional chaining for safe access
- Navbar: added optional chaining to handle undefined values safely
- Services: added optional chaining and conditional rendering
- Constants: added unique id for all data to be used as key in lists

Aeeshadam commented:
Thanks for the feedback @aayusachdev 

I have made all the suggested changes, kindly review at your earliest convenience.
Thanks

aayusachdev commented:
I see you `force-pushed` in this PR. What was the need of that?

aayusachdev reviewed (approved):
LGTM (Looks good to me)

Aeeshadam commented:
> I see you `force-pushed` in this PR. What was the need of that?

I had to force push because I needed to edit the commit message, I used the words "id" and "keys" in the message and after I pushed it, I realised it had changed those words to some weird strings(not sure where they came from). I understand it's best to avoid force push as much as possible and will avoid it going forward.
Thanks
